src/v/serde/avro/parser.cc

```diff
@@ -0,0 +1,170 @@
+#include "serde/avro/parser.h"
+
+#include "bytes/iobuf_parser.h"
+
+#include <seastar/util/defer.hh>
+
+namespace serde::avro {
+static constexpr int max_nested_depth = 200;
```

Comment:
Is this arbitrary or taken from somewhere? If arbitrary, consider reducing to 100 to match protobuf nesting limits.